src/aap_eda/api/vault.py

```diff
@@ -46,17 +46,27 @@ def encrypt_string(password: str, plaintext: str, vault_id: str) -> str:
     child.expect("Reading plaintext input from stdin*")
     child.sendline(plaintext)
     child.sendcontrol("D")
-    i = child.expect(["Encryption successful", "ERROR"])
-    if i == 0:
-        child.readline()
-        return "".join(
-            line.decode().lstrip()
-            for line in child
-            if not line.decode().startswith("!vault")
-        )
+
+    encrypted_successful_found = False
+    encrypted_start = False
+    encrypted_str = ""
+
+    for line in child.readlines():
```

Comment:
Can you explain the logic of this block? I don't understanding it.  According to my experience this would be the output of ansible-vault:
```
➜  ansible-vault encrypt_string --vault-id eda@test
Reading plaintext input from stdin. (ctrl-d to end input, twice if your content does not already have a newline)
hola
Encryption successful
!vault |
          $ANSIBLE_VAULT;1.2;AES256;eda
          32626239383264306562303631613137376633366434623035343238643462323036613765626162
          3932626337356539663933653036363162626665373039320a626139643331306564653231636563
          66643735663734386637323235313331323962336464366661373730333566373166316562326639
          3438366636633733330a303461303366313131393635363561623233353837356266636464666333
          3530
```

So it would be just as checking the return code and two lines before the encrypted text: "Encryption successful", and "!vault".

Reply:
What if we apply different logic for 2.14 or 2.16 instead? Not even I'm sure if we should support 2.14, since 2.16 will be the version of ansible for aap 2.5

Reply:
The new logic works for the 2 different versions of ansible-core which I feel is better than the old logic which was expecting the "Encryption successful" before the encrypted data.

Reply:
I meant encapsulate them to see easily which is the logic for 2.14 and which for 2.16. Or at least add a comment indicating it.

Reply:
@Alex-Izquierdo Comments added